### PR TITLE
fix: rewrite of menu component

### PR DESCRIPTION
**Pull request by Trancever**
### Motivation

Menu was broken on iOS, implementation was really complex and animations didn't use native driver.

### Test plan

Example

### CODE REVIEW

**Comment by cltsang** [comment]
Thanks so much for the quick fix.
I'd like to make a [dropdown menu](https://material.io/design/components/menus.html#dropdown-menu) like the one in the material design doc. Normally the menu should be anchored to the overflow button right?
By default, the menu is rendered covering the anchor, like in your gif. So I added marginTop 40 to the menu component's style. However, the margin part does not respond to touch to dismiss the menu.
It seems [the absolute fill](https://github.com/callstack/react-native-paper/pull/976/files#diff-c46d8ddb9464108a35b17099c084ea07R307) under it does not receive the touch event. Maybe because the menu's margin is wrapped by [another view,](https://github.com/callstack/react-native-paper/pull/976/files#diff-c46d8ddb9464108a35b17099c084ea07R310) and it captures the touch event?

So what should I do to do it properly? Thanks.

**Comment by zerobytes** [comment]
I'm facing this too. How long so we get the merge?

**Comment by mihaidaviddev** [comment]
@Trancever If you have the time, #977 might be related. Thank you for your work fixing this bug!

**Comment by Trancever** [comment]
@mihaidaviddev I am aware of this issue and yes, this PR will fix it.

**Comment by Trancever** [comment]
@cltsang With this PR, passing padding in style will be supported.

**Comment by cltsang** [comment]
Thank you so much!

File: src/components/Menu/Menu.js [hunk 1]
```diff
@@ -124,188 +128,281 @@ class Menu extends React.Component<Props, State> {
...
-        const left = Math.max(SCREEN_INDENT, x);
-
-        this.setState({ menuState: 'shown', top, left });
-      });
-    }
-  };
-
   _hide = () => {
-    BackHandler.removeEventListener('hardwareBackPress', this._hide);
+    BackHandler.removeEventListener('hardwareBackPress', this.props.onDismiss);
```
Comment (ferrannp) [inline src/components/Menu/Menu.js]: Can you call this in `componentWillUnmount` too? If they for example navigate outside the screen with the menu opened, we are gonna make back button not operative (it happened with Modal).

Comment (Trancever) [reply src/components/Menu/Menu.js]: Sure, is there anything else to address or we can proceed with merge once this one is fixed?

Comment (ferrannp) [reply src/components/Menu/Menu.js]: For me it's good!

**Comment by iyadthayyil** [comment]
I am so sorry for leaving #485 open for a long time.
I couldn't do any development  in past few months.
Thanks @waquidvp 
Thank you very much @Trancever for making it perfect!

**Comment by ferrannp** [comment]
Thanks for your work @iyadthayyil !

**Comment by Trancever** [comment]
@iyadthayyil Without your work, it wouldn't be possible, thanks!

**Pull request merged**